test/patch/requestAnimationFrame.spec.js

```diff
@@ -3,63 +3,82 @@
-      testZone.run(function() {
-        if (typeof mozRequestAnimationFrame === 'undefined') {
-          console.log('WARNING: skipping mozRequestAnimationFrame test (missing this API)');
-          return done();
-        }
-
-        // Some browsers (especially Safari) do not fire mozRequestAnimationFrame
-        // if they are offscreen. We can disable this test for those browsers and
-        // assume the patch works if setTimeout works, since they are mechanically
-        // the same
-        mozRequestAnimationFrame(function () {
-          expect(zone).toBeDirectChildOf(testZone);
+      it('should execute in the current zone if it is not the root', function (done) {
+        testZone.run(function () {
+          rAF(function () {
+            expect(zone === testZone).toBe(true);
+            done();
+          });
+        });
+      });
+      it('should fork a new zone if run from the root zone', function (done) {
+        expect(zone.isRootZone()).toBe(true);
+        rAF(function () {
+          expect(zone.parent.isRootZone()).toBe(true);
           done();
         });
       });
-    });
-  });
 
-  describe('webkitRequestAnimationFrame', function () {
-    it('should run the passed callback in a zone', function (done) {
-      testZone.run(function() {
-        if (typeof webkitRequestAnimationFrame === 'undefined') {
-          console.log('WARNING: skipping webkitRequestAnimationFrame test (missing this API)');
-          return done();
-        }
+      it('should bind to parent zone when called recursively', function (done) {
+        testZone.run(function () {
+          var frames = 0;
 
-        // Some browsers (especially Safari) do not fire webkitRequestAnimationFrame
-        // if they are offscreen. We can disable this test for those browsers and
-        // assume the patch works if setTimeout works, since they are mechanically
-        // the same
-        webkitRequestAnimationFrame(function () {
-          expect(zone).toBeDirectChildOf(testZone);
-          done();
+          function frameCallback() {
+            expect(zone === testZone).toBe(true);
+            if (frames++ > 15) {
+              return done();
+            }
+            rAF(frameCallback);
+          }
+
+          rAF(frameCallback);
         });
       });
+
+      //it('should bind to parent zone when called recursively in an async fashion', function (done) {
+      //  testZone.run(function () {
+      //    var frames = 0;
+      //
+      //    function frameCallback() {
+      //      // For this check, since we're injecting ANOTHER zone by using
+      //      // setTimeout, we need to expect that the parent zone will be
+      //      // our test zone.
+      //      var expectedZone = frames ? zone.parent : zone;
+      //      expect(expectedZone === testZone).toBe(true);
+      //      if (frames++ > 15) {
+      //        return done();
+      //      }
+      //      setTimeout(function () {
```

Comment:
Hmm, I don't see the original solution being much simpler, and it definitely fails in this case.

The reason I'm concerned with this issue is that Zone.js is wrapping all code executed in an angular2 app, and given the popularity of angular, I would assume you'll get every kind of bone-headed app setup that is conceivable.  

It's not my project though, so I'll revert this back to what it was originally and remove the test this evening.